File: army/src/main/java/com/rolan/army/ApplicationLifecycle.java
package com.rolan.army;

import androidx.annotation.NonNull;

/**
 * Created by Rolan.Wang on 1/28/21.6:58 PM
 * 描述：
 */
class ApplicationLifecycle implements Lifecycle {
    @Override
    public void addListener(@NonNull LifecycleListener listener) {
        listener.onStart();
    }

    @Override
    public void removeListener(@NonNull LifecycleListener listener) {
        // Do nothing.
    }
}

File: army/src/main/java/com/rolan/army/ApplicationLifecycle.kt
package com.rolan.army

/**
 * Created by Rolan.Wang on 1/28/21.6:58 PM
 * application Lifecycle
 */
internal class ApplicationLifecycle : Lifecycle {
    override fun addListener(listener: LifecycleListener) {
        listener.onStart()
    }

    override fun removeListener(listener: LifecycleListener) {
        // Do nothing.
    }
}